src/odemis/util/transform.py

```diff
@@ -0,0 +1,342 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 29 Nov 2013
+
+@author: Kimon Tsitsikas
+
+Copyright © 2012-2013 Kimon Tsitsikas, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms  of the GNU General Public License version 2 as published by the Free
+Software  Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY;  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR  PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+
+from __future__ import division
+
+import numpy
+from numpy.linalg import LinAlgError
+import scipy.optimize
+
+from odemis.util.linalg import tri_inv
+
+
+class Transform(object):
+    """
+    A Transform class that is useful for geometrical coordinate transformations.
+    """
+
+    def __init__(self, rotation=0, scaling=1, shear=0, translation=numpy.zeros(2)):
```

Comment:
You should not initialise with `numpy.zeros(2)`, because it'd be an object shared by _all_ the instances, and chaos would ensue. So, have `None` by default, and in the code, replace `None` by `numpy.zeros(2)`.
See https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument for more info.